client/src/components/classesPage/components/Students.js

```diff
@@ -65,53 +79,95 @@ function students(props) {
     props.setSelectedStudents(updatedStudents);
   }
 
+  function editStudent(id) {
+    console.log(id);
+    const [student] = props.students.filter(s => s.student_id === id);
+    student.isEditing = true;
+    console.log(student);
+  }
+  // <Checkbox
+  //   color="secondary"
+  //   value={`${s.student_id}`}
+  //   checked={props.selectedStudents.includes(s.student_id)}
+  //   onClick={e => selectStudent(e)}
+  // />
+
+  function handleChange(e, student) {
+    console.log(e.target.value);
+    console.log(student);
+    console.log(e.target.name);
+    student[e.target.name] = e.target.value;
+  }
+
   return (
     <>
       <Typography variant="h6" className={classes.title} gutterBottom>
         Current Students
       </Typography>
       <Card className={classes.studentList}>
-        {props.students.map((s, i) => (
-          <Grid key={i}>
-            <span>{`${s.name}`}</span>
-            <Checkbox
-              color="secondary"
-              value={`${s.student_id}`}
-              checked={props.selectedStudents.includes(s.student_id)}
-              onClick={e => selectStudent(e)}
-            />
-          </Grid>
-        ))}
-        {addedStudents.map((s, i) => (
-          <Grid key={i}>
-            <span style={{ fontWeight: 'bold' }}>{`${s.first_name} ${
-              s.last_name
-            }`}</span>
-            <Checkbox
-              color="primary"
-              value={`${s.student_id}`}
-              checked={props.selectedStudents.includes(s.student_id)}
-              onClick={e => selectStudent(e)}
-            />
+        {props.students.map(s => (
+          <Grid key={s.student_id}>
+            <span>{`${s.first_name} ${s.last_name} `}</span>
+            <Create onClick={e => props.toggleEditStudent(s)} />
+            {s.isActiveStudent && (
```

Comment:
Good job w/ this conditional rendering, I think at the higher levels of react conditional rendering comes way more into play.